MaslowDue/report.cpp

```diff
@@ -46,69 +46,47 @@ static void report_util_axis_values(float *axis_value) {
-  serial_write(' ');
-  serial_write('(');
   switch(n) {
-    case 0: printPgmString(PSTR("stp pulse")); break;
-    case 1: printPgmString(PSTR("idl delay")); break; 
-    case 2: printPgmString(PSTR("stp inv")); break;
-    case 3: printPgmString(PSTR("dir inv")); break;
-    case 4: printPgmString(PSTR("stp en inv")); break;
-    case 5: printPgmString(PSTR("lim inv")); break;
-    case 6: printPgmString(PSTR("prb inv")); break;
-    case 10: printPgmString(PSTR("rpt")); break;
-    case 11: printPgmString(PSTR("jnc dev")); break;
-    case 12: printPgmString(PSTR("arc tol")); break;
-    case 13: printPgmString(PSTR("rpt inch")); break;
-    case 20: printPgmString(PSTR("sft lim")); break;
-    case 21: printPgmString(PSTR("hrd lim")); break;
-    case 22: printPgmString(PSTR("hm cyc")); break;
-    case 23: printPgmString(PSTR("hm dir inv")); break;
-    case 24: printPgmString(PSTR("hm feed")); break;
-    case 25: printPgmString(PSTR("hm seek")); break;
-    case 26: printPgmString(PSTR("hm delay")); break;
-    case 27: printPgmString(PSTR("hm pulloff")); break;
-    case 30: printPgmString(PSTR("rpm max")); break;
-    case 31: printPgmString(PSTR("rpm min")); break;
-    case 32: printPgmString(PSTR("laser")); break;
-    default:
-      n -= AXIS_SETTINGS_START_VAL;
-      uint8_t idx = 0;
-      while (n >= AXIS_SETTINGS_INCREMENT) {
-        n -= AXIS_SETTINGS_INCREMENT;
-        idx++;
-      }
-      serial_write(n+'x');
-      switch (idx) {
-        case 0: printPgmString(PSTR(":stp/mm")); break;
-        case 1: printPgmString(PSTR(":mm/min")); break;
-        case 2: printPgmString(PSTR(":mm/s^2")); break;
-        case 3: printPgmString(PSTR(":mm max")); break;
-      }
-      break;
+#ifdef MASLOWCNC
+    case GRBL_CHAIN_OVER_SPROCKET: printPgmString(PSTR(" (chain over sprocket)")); break;
+    case GRBL_MACHINE_WIDTH: printPgmString(PSTR(" (machine width)")); break;
+    case GRBL_MACHINE_HEIGHT: printPgmString(PSTR(" (machine height)")); break;
+    case GRBL_DIST_BETWEEN_MOTORS: printPgmString(PSTR(" (dist between motors)")); break;
+    case GRBL_MOTOR_OFFSET_Y: printPgmString(PSTR(" (motor offset y)")); break;
+    case GRBL_X_CORR_SCALING: printPgmString(PSTR(" (x corr scaling)")); break;
+    case GRBL_Y_CORR_SCALING: printPgmString(PSTR(" (y corr scaling)")); break;
+    case GRBL_CHAIN_SAG_CORRECTION: printPgmString(PSTR(" (chain sag correction)")); break;
+    case GRBL_LEFT_CHAIN_TOLERANCE: printPgmString(PSTR(" (left chain tolerance)")); break;
+    case GRBL_RIGHT_CHAIN_TOLERANCE: printPgmString(PSTR(" (right chain tolerance)")); break;
+    case GRBL_ROTATION_DISK_RADIUS: printPgmString(PSTR(" (rotation disk radius)")); break;
+    case GRBL_CHAIN_LENGTH: printPgmString(PSTR(" (chain length)")); break;
+    case GRBL_Z_TRAVEL_MIN: printPgmString(PSTR(" (Z-axis maximum travel, millimeters)")); break;
+    case GRBL_KINEMATICS_MODE: printPgmString(PSTR(" (kinematics mode, 0=forward, 1=triangular)")); break;
```

Comment:
The two different Kinematics modes are triangular and quadrilateral depending on if a user has a ring or linkage (triangular) or the old school original rigid sled (quadrilateral).

Forward and inverse refer to the position calculations.  Forward (chain to position) and Inverse (position to chain).

Reply:
Good point. Since quad. is not implemented in this firmware, it sounds like what I want is to change this to a boolean `GRBL_KINEMATICS_FORWARD`, which would then be suitable for quad. kinematics (or other).